sandpack-react/src/themes/index.ts

```diff
@@ -175,9 +176,18 @@ export const SANDPACK_THEMES: Record<SandpackPredefinedTheme, SandpackTheme> = {
   'monokai-pro': monokaiProTheme,
 };
 
-export const createThemeObject = (
-  inputTheme?: SandpackPredefinedTheme | SandpackPartialTheme
-) => {
+export const createThemeObject = (inputTheme?: SandpackThemeProp) => {
+  if (inputTheme === 'auto') {
+    const theme = getDarkModePreference()
```

Comment:
here you can also assign the id to `codesandbox-light` or `codesandbox-dark` so you don't generate a new hash for the theme. `auto` will still set a standard theme, the hash function is created for custom themes. Also, move this after the `if(inputTheme === undefined)`